crates/biome_js_parser/tests/js_test_suite/error/import_type_error.ts.snap

```diff
@@ -0,0 +1,393 @@
+                  2: R_BRACK@78..79 "]" [] []
+                  3: COLON@79..80 ":" [] []
+                  4: JS_STRING_LITERAL@80..83 "\"1\"" [] []
+              4: R_CURLY@83..84 "}" [] []
+            2: R_CURLY@84..85 "}" [] []
+          4: R_PAREN@85..86 ")" [] []
+        3: (empty)
+        4: (empty)
+      5: SEMICOLON@86..87 ";" [] []
+    5: TS_TYPE_ALIAS_DECLARATION@87..129
+      0: TYPE_KW@87..93 "type" [Newline("\n")] [Whitespace(" ")]
+      1: TS_IDENTIFIER_BINDING@93..95
+        0: IDENT@93..95 "C" [] [Whitespace(" ")]
+      2: (empty)
+      3: EQ@95..97 "=" [] [Whitespace(" ")]
+      4: TS_IMPORT_TYPE@97..129
+        0: TYPEOF_KW@97..104 "typeof" [] [Whitespace(" ")]
+        1: IMPORT_KW@104..110 "import" [] []
+        2: TS_IMPORT_TYPE_ARGUMENTS@110..129
+          0: L_PAREN@110..111 "(" [] []
+          1: TS_STRING_LITERAL_TYPE@111..119
+            0: JS_STRING_LITERAL@111..119 "\"a.json\"" [] []
+          2: COMMA@119..120 "," [] []
+          3: TS_IMPORT_TYPE_ASSERTION_BLOCK@120..129
+            0: L_CURLY@120..121 "{" [] []
+            1: TS_IMPORT_TYPE_ASSERTION@121..128
+              0: WITH_KW@121..125 "with" [] []
+              1: COLON@125..126 ":" [] []
+              2: L_CURLY@126..127 "{" [] []
+              3: JS_IMPORT_ASSERTION_ENTRY_LIST@127..127
+              4: R_CURLY@127..128 "}" [] []
+            2: R_CURLY@128..129 "}" [] []
+          4: (empty)
+        3: (empty)
+        4: (empty)
+      5: (empty)
+    6: JS_BOGUS_STATEMENT@129..133
+      0: COMMA@129..131 "," [] [Whitespace(" ")]
+      1: JS_NUMBER_LITERAL@131..132 "1" [] []
+      2: R_PAREN@132..133 ")" [] []
+    7: JS_EMPTY_STATEMENT@133..134
+      0: SEMICOLON@133..134 ";" [] []
+    8: TS_TYPE_ALIAS_DECLARATION@134..157
+      0: TYPE_KW@134..140 "type" [Newline("\n")] [Whitespace(" ")]
+      1: TS_IDENTIFIER_BINDING@140..142
+        0: IDENT@140..142 "D" [] [Whitespace(" ")]
+      2: (empty)
+      3: EQ@142..144 "=" [] [Whitespace(" ")]
+      4: TS_IMPORT_TYPE@144..156
+        0: (empty)
+        1: IMPORT_KW@144..150 "import" [] []
+        2: TS_IMPORT_TYPE_ARGUMENTS@150..156
+          0: L_PAREN@150..151 "(" [] []
+          1: TS_STRING_LITERAL_TYPE@151..154
+            0: JS_STRING_LITERAL@151..154 "\"a\"" [] []
+          2: COMMA@154..155 "," [] []
+          3: (empty)
+          4: R_PAREN@155..156 ")" [] []
+        3: (empty)
+        4: (empty)
+      5: SEMICOLON@156..157 ";" [] []
+  4: EOF@157..158 "" [Newline("\n")] []
+
+```
+
+## Diagnostics
+
+```
+import_type_error.ts:1:33 parse ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
+
+  × Missing import type assertion keyword 'with'
+  
+  > 1 │ type A = typeof import("a.json",{});
+      │                                 ^
+    2 │ type B = typeof import("a.json",{with:{[a]:"1"}});
+    3 │ type C = typeof import("a.json",{with:{}}, 1);
+  
+  i 'with' expected.
+  
+  > 1 │ type A = typeof import("a.json",{});
```

Comment:
nit: The range here is incorrect, because `with` must be after `{`, which means that `with` should be where `}`. Hence we should check for the next token, IMHO

Reply:
yes, Typescript compiler do the same thing, i will update later